### PR TITLE
Add functional calls to the cli

### PR DESCRIPTION
**Pull request by bmesuere**
_[Original pull request](https://github.ugent.be/unipept/unipept-cli/pull/55) by @bmesuere on Thu May 23 2019 at 10:29._
_Merged by @bmesuere on Wed Jun 19 2019 at 09:54._

### CODE REVIEW

File: lib/formatters.rb [hunk 1]
```diff
@@ -186,7 +186,22 @@ def header(data, fasta_mapper = nil)
       CSV.generate do |csv|
         first = data.first
         keys = fasta_mapper ? ['fasta_header'] : []
-        csv << (keys + first.keys).map(&:to_s) if first
+
+        firstKeys = first.keys
+
+        if firstKeys.include?("ec")
```
Comment (bmesuere) [inline lib/formatters.rb]: I would deduplicate these 2 if's and wrap them in something like
```ruby
['ec', 'go'].each do |annotation|
   code
end
```

_[Original comment](https://github.ugent.be/unipept/unipept-cli/pull/55#discussion_r21549) by @bmesuere on Thu May 23 2019 at 10:36._

File: lib/formatters.rb [hunk 2]
```diff
@@ -186,7 +186,22 @@ def header(data, fasta_mapper = nil)
       CSV.generate do |csv|
         first = data.first
         keys = fasta_mapper ? ['fasta_header'] : []
-        csv << (keys + first.keys).map(&:to_s) if first
+
+        firstKeys = first.keys
```
Comment (bmesuere) [inline lib/formatters.rb]: Can't we reuse the `keys` variable for this? Something like `keys += first.keys`?

_[Original comment](https://github.ugent.be/unipept/unipept-cli/pull/55#discussion_r21550) by @bmesuere on Thu May 23 2019 at 10:46._

File: lib/formatters.rb [hunk 3]
```diff
@@ -204,7 +219,17 @@ def footer
...
         data.each do |o|
-          csv << o.values.map { |v| v == '' ? nil : v }
+          row = []
+          o.each do |k, v|
+            if k == "ec" || k == "go"
+              v.first.keys.each do |key|
+                row << (v.map { |el| el[key] }).join(" ")
+              end
+            else
+              row << (v == '' || k == "ec" || k == "go" ? nil : v)
```
Comment (bmesuere) [inline lib/formatters.rb]: Why the ec and go check here? Aren't those handled by the initial `if` check?

_[Original comment](https://github.ugent.be/unipept/unipept-cli/pull/55#discussion_r21551) by @bmesuere on Thu May 23 2019 at 10:53._

**Comment by pverscha** [comment]
Thanks! I've updated the files according to the feedback. I've changed one more thing. Up until now, the CSV-header stated: `peptide,total_protein_count,ec_number,protein_count`. I've changed the header so that `protein_count` now says `ec_protein_count` and `go_protein_count` respectively.

_[Original comment](https://github.ugent.be/unipept/unipept-cli/pull/55#issuecomment-90597) by @pverscha on Thu May 23 2019 at 12:48._

**Pull request merged**